runtime/caml/platform.h

```diff
@@ -111,8 +111,11 @@ Caml_inline void cpu_relax(void) {
 
 #ifdef _WIN32
 
-typedef SRWLOCK caml_plat_mutex;
-#define CAML_PLAT_MUTEX_INITIALIZER SRWLOCK_INIT
+typedef struct {
+  SRWLOCK lock;
+  _Atomic DWORD owner_tid;      /* 0 if not owned */
```

Comment:
There's a clarifying comment from the 4.x implementation, and I think it's worth having an updated version here:
https://github.com/ocaml/ocaml/blob/8899c05cfc9027e74dfe88c71cc78bac5102e771/otherlibs/systhreads/st_win32.h#L206-L212

Reply:
I've had a look to winpthreads and ntpl, only winpthreads marks this field as `volatile`, none of them use any sort of lock or atomic access to read or write to it. I'll just add:
> The "owner_tid" field is not always protected by "lock"; it is also accessed without holding "lock".